internal/api_client/sddc_manager_client.go

```diff
@@ -205,25 +189,21 @@ func (sddcManagerClient *SddcManagerClient) GetResourceIdAssociatedWithTask(ctx
 	if err != nil {
 		return "", err
 	}
-	if len(task.Resources) == 0 {
+	if len(*task.Resources) == 0 {
 		return "", fmt.Errorf("no resources associated with Task with ID %q", taskId)
 	}
-	for _, resource := range task.Resources {
-		if *resource.Type == resourceType {
-			return *resource.ResourceID, nil
+	for _, resource := range *task.Resources {
+		if resource.Type == resourceType {
+			return resource.ResourceId, nil
 		}
 	}
 	return "", fmt.Errorf("task %q did not contain resources of type %q", taskId, resourceType)
 }
 
-func (sddcManagerClient *SddcManagerClient) getTask(ctx context.Context, taskId string) (*models.Task, error) {
+func (sddcManagerClient *SddcManagerClient) getTask(ctx context.Context, taskId string) (*vcf.Task, error) {
 	apiClient := sddcManagerClient.ApiClient
-	getTaskParams := tasks.NewGetTaskParamsWithTimeout(constants.DefaultVcfApiCallTimeout).
-		WithContext(ctx)
-	getTaskParams.ID = taskId
-
-	getTaskResult, err := apiClient.Tasks.GetTask(getTaskParams)
-	if err != nil {
+	res, err := apiClient.GetTaskWithResponse(ctx, taskId)
+	if err != nil || res.StatusCode() != 200 {
```

Comment:
Eww! This generated client leaks http abstraction outside

Reply:
Yes this is one of the downsides of this generator